staging/src/k8s.io/apiserver/pkg/cel/library/cost.go

```diff
@@ -421,26 +421,24 @@ func (l *CostEstimator) EstimateCallCost(function, overloadId string, target *ch
 			return &checker.CallEstimate{CostEstimate: strCost.Multiply(regexCost), ResultSize: &checker.SizeEstimate{Min: 0, Max: sz.Max}}
 		}
 	case "cidr", "isIP", "isCIDR":
-		if target != nil {
+		if target == nil {
```

Comment:
do we need to check target at all? I'm having a hard time understanding how the logic flipped

Reply:
Thanks for catching it. I removed the target checking logic and it should work fine with either path

Reply:
There was a long-standing bug which was not tickled until I added list cost-tracking.  The `*target` arg was initialized as `var target AstNode` upstream and passed to implementations as `&target`. This resulted in a pointer to the empty interface rather than a pointer to `nil`. The assertion of a non-nil target resulted in a no-op as a consequence.

In short, you could remove the reference to `target != nil` and it would not have changed anything in the older code, but I did note one other minor logic error and fixed it as well